### PR TITLE
Tests for models

### PR DESCRIPTION
**Pull request by ChristianChidiac**
Added tests for the following models: User.java, Quiz.java, RecognitionQuiz.java, Artist.java, and Track.java

### CODE REVIEW

File: src/main/java/com/group6/harmoniq/models/Quiz.java [hunk 1]
```diff
@@ -25,6 +25,10 @@ public Quiz(String question_path, String answer, String option1, String option2,
     }
     public Long getId() {
         return id;
+    }
+     //For testing purposes
+     public void setId(Long id) {
```
Comment (sbaljeet13) [inline src/main/java/com/group6/harmoniq/models/Quiz.java]: Adding a new methond or exposing a readonly field just for testing is usually not a good approach. The tests should be adapted to the model or unit of code that we are trying to test, not other way around. In this case, we can skip the getId test if its not possible without adding this method to the model

File: src/main/java/com/group6/harmoniq/models/RecognitionQuiz.java [hunk 2]
```diff
@@ -27,6 +27,10 @@ public RecognitionQuiz(String preview_url, String answer, String option1, String
     public Long getId() {
         return id;
     }
+    //For testing purposes
+    public void setId(Long id) {
```
Comment (sbaljeet13) [inline src/main/java/com/group6/harmoniq/models/RecognitionQuiz.java]: This should also be removed

File: src/main/java/com/group6/harmoniq/models/User.java [hunk 3]
```diff
@@ -69,6 +69,11 @@ public User(String spotifyId, String displayName, String email, int followers,
     public long getUId() {
         return uid;
     }
+    
+    //For testing purposes
+    public void setUId(Long uid) {
```
Comment (sbaljeet13) [inline src/main/java/com/group6/harmoniq/models/User.java]: And this as well

File: src/test/java/com/group6/harmoniq/models/ArtistTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.springframework.util.Assert;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/ArtistTest.java]: I think these are unused and can be removed
```suggestion

```

File: src/test/java/com/group6/harmoniq/models/QuizTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.springframework.util.Assert;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/QuizTest.java]: These are also unused

File: src/test/java/com/group6/harmoniq/models/RecognitionQuizTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.springframework.util.Assert;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/RecognitionQuizTest.java]: These also seem unused

File: src/test/java/com/group6/harmoniq/models/TrackTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.springframework.util.Assert;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/TrackTest.java]: Some of these are also unused

File: src/test/java/com/group6/harmoniq/models/UserTest.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
+package com.group6.harmoniq.models;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/UserTest.java]:
```suggestion

```

File: src/test/java/com/group6/harmoniq/models/UserTest.java [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.util.Assert;
```
Comment (sbaljeet13) [inline src/test/java/com/group6/harmoniq/models/UserTest.java]:
```suggestion

```

**Review by sbaljeet13 — CHANGES_REQUESTED** [review]

**Review by sbaljeet13 — APPROVED** [review]

**Review by yBoranbayev — APPROVED** [review]

**Pull request merged**